tests/amqp.spec.js

```diff
@@ -0,0 +1,190 @@
+import {expect} from 'chai';
+import {READERS} from '@natlibfi/fixura';
+import generateTests from '@natlibfi/fixugen';
+import * as fakeAmqpLib from '@onify/fake-amqplib';
+import {createAmqpOperator} from '../src/amqp.js';
+
+run();
+
+async function run() {
+  const amqpOperator = await createAmqpOperator(fakeAmqpLib, 'amqp://example.com');
+  const blobId = 'test-queue-aaa-111-bbb';
+  const status = 'PENDING_TEST';
+
+  generateTests({
+    callback,
+    path: [__dirname, '..', 'test-fixtures', 'amqp'],
+    recurse: false,
+    useMetadataFile: true,
+    fixura: {
+      failWhenNotFound: true,
+      reader: READERS.JSON
+    },
+    mocha: {
+      beforeEach: async () => {
+        const count = await amqpOperator.countQueue({blobId, status});
+        if (count > 0) {
+          throw new Error('TEST QUEUE CONTAMINATION! ', count);
+        }
+      },
+      afterEach: async () => {
+        await amqpOperator.purgeQueue({blobId, status});
+        await amqpOperator.removeQueue({blobId, status});
+      },
+      after: () => {
+        fakeAmqpLib.resetMock();
+      }
+    }
+  });
+
+  async function callback({
+    getFixture,
+    operationParams = {
+      queueRecords: 10,
+      operations: []
+    },
+    expectedToFail = false,
+    expectedErrorStatus = 200,
+    expectedErrorMessage = ''
+  }) {
+    try {
+      const prepared = await prepareQueues({blobId, status, ...operationParams});
+      const operationsDone = await runOperations({blobId, status, ...operationParams});
+      console.log(operationsDone); // eslint-disable-line
+
+      const count = await amqpOperator.countQueue({blobId, status, ...operationParams});
+
+      const expectedResult = await getFixture('expectedResult.json');
+      expect({prepared, operationsDone, count}).to.eql(expectedResult);
+    } catch (error) {
+      if (!expectedToFail) {
+        throw error;
+      }
+
+      // console.log(error); // eslint-disable-line
+      expect(error.status).to.eql(expectedErrorStatus);
+      expect(error.payload).to.eql(expectedErrorMessage);
+      expect(expectedToFail).to.eql(true, 'This test is not suppose to fail!');
+    }
+
+    async function prepareQueues({blobId, status, queueRecords}) {
+      if (queueRecords > 0) {
+        await amqpOperator.countQueue({blobId, status});
```

Comment:
Why countQueue here?

Reply:
count is made with amqplibs assert queue method (https://amqp-node.github.io/amqplib/channel_api.html#channel_assertQueue) that will make sure queue is existing.